### ISSUE TITLE
Serializer generated with @Serializer always encodes null, ignores "encodeDefaults = false"

### ISSUE THREAD

**Issue by dimsuz** [Kotlin/kotlinx.serialization#1268]
**Describe the bug**
I have a `@Serializer` annotation to generate me a serializer (used for delegation which is not included in example, as in #1253).
Turns out such serializer ignores `enableDefaults = false` option and always encodes null values, even default ones.
While a "default" serializer generated by plugin works correctly.

**To Reproduce**
```kotlin
@Serializer(forClass = Test::class)
object TestDefaultSerializer

@Serializable(with = TestDefaultSerializer::class)
data class Test(
  val x: Int,
  val text: String? = null
)

// and now
Json { encodeDefaults = false }.encodeToString(Test(33))
```
encodes to
```
{"x":33,"text":null}
```

But if I use plugin-generated serializer as in

```kotlin
@Serializable
data class Test(
  val x: Int,
  val text: String? = null
)

// and now
Json { encodeDefaults = false }.encodeToString(Test(33))
```
encodes to
```
{"x":33}
```

**Expected behavior**
Serializer generated with `@Serializer` behaves the same as "default" serializer and does not encode nulls for default properties.

**Environment**
 - Kotlin version: 1.4.21
 - Library version: 1.0.1
 - Kotlin platforms: JVM
 - Gradle version: 6.5

**Comment by sandwwraith** [comment]
That's a functionality that is currently unsupported in external serialization. cc @shanshin

**Comment by shanshin** [comment]
Fixed since kotlin 1.5.10
